utils/exec_test.go

```diff
@@ -30,24 +31,53 @@ func TestExec(t *testing.T) {
 	})
 
 	t.Run("bad command", func(t *testing.T) {
-		stdout, stderr, err := exec(nil, `bad_command`, true)
+		stdout, stderr, err := exec(nil, nil, `bad_command`, true)
 		assert.Error(t, err)
 		assert.Empty(t, stdout)
 		assert.Contains(t, stderr, "not found")
 	})
 
 	t.Run("print to stderr", func(t *testing.T) {
-		stdout, stderr, err := exec(t, `echo "failing" >&2`, true)
+		stdout, stderr, err := exec(t, nil, `echo "failing" >&2`, true)
 		assert.NoError(t, err)
 		assert.Empty(t, stdout)
 		assert.Equal(t, "failing\n", stderr)
 	})
 
 	t.Run("stderr then stdout", func(t *testing.T) {
-		stdout, stderr, err := exec(t, `echo "failing" >&2; echo "succeeding"`, true)
+		stdout, stderr, err := exec(t, nil, `echo "failing" >&2; echo "succeeding"`, true)
 		assert.NoError(t, err)
 		assert.Equal(t, "failing\n", stderr)
 		assert.Equal(t, "succeeding\n", stdout)
 	})
 
+	t.Run("context with cancel", func(t *testing.T) {
+		timeout := 5 * time.Second
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		defer cancel()
```

Comment:
Defer should not be used in tests that can fail and exit without returning. In such cases, `cancel()` will not be called. See instead https://pkg.go.dev/testing#T.Cleanup